iso8601/parse.go

```diff
@@ -0,0 +1,106 @@
+package iso8601
+
+import (
+	"encoding/binary"
+	"time"
+	"unsafe"
+)
+
+// Parse parses an ISO8601 string, e.g. "2021-03-25T21:36:12Z".
+func Parse(input string) (time.Time, error) {
+	switch len(input) {
+	case 20:
+		b := unsafeStringToBytes(input)
+		_ = b[19] // compiler hint
```

Comment:
Can we remove this if we convert the input to bytes at the beginning of the function, then switch on `len(b)` ? (then the compiler already knows the byte slice has length 20)